### ISSUE TITLE
[bug] virtualenv generator strip '\' on windows when upgrading to conan 1.21.0

### ISSUE THREAD

**Issue by Thiel** [conan-io/conan#6301]
Hi,

After updating to 1.21.0, our CI and developer have been experiencing issues on script and tool that relied on conan generated activate.sh/activate_run.sh on windows. It looks like that it may be caused by some recent change on those generator.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version:
  * Conan version: 1.21.0
  * Python version: 2.7.12

### Steps to reproduce (Include if Applicable)

Here's the small reproduce step I could come up with

```
conan install cmake_installer/3.15.4@conan/stable -g virtualenv -if build
source build/activate.sh
echo $CMAKE_ROOT
```
### Logs (Executed commands with output) (Include/Attach if Applicable)

With conan 1.21.0
```
$ echo $CMAKE_ROOT
C:Usersjbounthong.conandatacmake_installer3.15.4conanstablepackage60b04a80ac4dd5c23f1acffacc679e9902e636e5
```

With conan 1.20.5
```
$ echo $CMAKE_ROOT
C:\Users\jbounthong\.conan\data\cmake_installer\3.15.4\conan\stable\package\60b04a80ac4dd5c23f1acffacc679e9902e636e5
```

### Cause & Impact

The regression seems to be cause by recent change on virtualenv, virtualrunenv generator, and the introduction of `environment.sh.env`. When it's being sourced by the `activate.sh` the windows `\` directory separators are being stripped down.

We are being impacted by it since we rely on activate_run.sh to set the PATH variable so that built `.exe` execution can lookup dependent `.dll` provided by conan.

**Comment by Thiel** [comment]
With some digging, seems like on line https://github.com/conan-io/conan/blame/develop/conans/client/generators/virtualenv.py#L18

`read` command should be used with `-r` option to preserve the `\` characters.

According to https://ss64.com/bash/read.html
```
   -r        Backslash  does not act as an escape character.  The backslash is considered to be part
             of the line. In particular, a backslash-newline pair can not be used as a line continuation.
```

Quick experimentation by modifying the generated activate script seems to suggest that would fix the issue.

**Comment by kucerad** [comment]
Met this issue today as well... We run the generator on Windows but use the activate.sh in the bash (git-bash) and the backslashes cause exactly the issues described here. I don't see any reason why .sh file should contain "\" in paths. This is definitely a bug and needs to be fixed...

**Comment by kucerad** [comment]
Guys, this need to be handled more carefully than @Thiel is suggesting. Sure, the backslashes need to be preserved, but still can cause issues because we'd need standard slashes in the Unix paths instead! 
Suppose this:
conanfile.py contains
`def package_info(self):
        self.env_info.conan_package_path = self.package_folder`
Then we'd like to export the activate.sh to be used in Windows for git pre-commit hook (using git-bash where the paths should contain the standard Unix path-slash)... See the issue?

**Comment by memsharded** [comment]
Hi,

Managed to reproduce the issue, indeed it seems it should be fixed.

@kucerad Can you please elaborate a bit more about the issue, not sure how ``self.env_info.conan_package_path = self.package_folder`` will be different to what is happening now.

**Comment by jgsogo** [comment]
Hi @Thiel ! Thanks a lot for taking the time to investigate the issue and drill down to the PR, the line of code, and a possible fix.

This is a bug, indeed. I think we don't have any test using `sh` scripts in Windows for virtualenvs, and it would be totally needed. If you are reproducing it, @memsharded, can you provide a minimal test to add it to our test suite? On top of that, I'll build the actual test case and think about any other corner case.

Thanks

**Comment by jgsogo** [comment]
For the CI we might be able to leverage on Git bash (probably installed together with Git) in order to run those scripts in the test, I've found this in SO, it can help (but I haven't tried it): https://stackoverflow.com/questions/21564275/windows-shortcut-to-run-a-git-bash-script

```
C:\WINNT\system32\cmd.exe /c ""C:\Git\bin\sh.exe" --login -i -- D:\temp\test.sh"
```

**Comment by kucerad** [comment]
@memsharded I just meant, that what was @Thiel suggesting might not fix the issue I'm observing... Basically paths need to be handled differently if the output is meant for Unix and for Windows. 
I just wanted to demonstrate, where I'm getting the path for my env. variable - (just to eliminate the possible source of issues on my side)...

**Comment by Thiel** [comment]
Hi,

@memsharded  & @jgsogo Thank you for jumping on the issue so quickly.
Sorry for not being able to contribute directly.
I'm still catching up with the contribution rules and setting up a suitable test environment, on windows it doesn't look trivial nonetheless it is still great to see the engineering quality that goes behind conan development.
Is there anything I can do to help? Or is the matter in good hand?

@kucerad It seems like you're concerned about the slashes orientation. Which may be an issue, but while using conan 2.20.5 with git-bash, the `activate.sh` seemed to be doing fine with `\` and `C:` in the path.

```
$ cat activate.sh
OLD_PS1="$PS1"
export OLD_PS1
PS1="(conanenv) $PS1"
export PS1
CMAKE_ROOT="C:\Users\jbounthong\.conan\data\cmake_installer\3.15.4\conan\stable\package\60b04a80ac4dd5c23f1acffacc679e9902e636e5"
export CMAKE_ROOT
CMAKE_MODULE_PATH="C:\Users\jbounthong\.conan\data\cmake_installer\3.15.4\conan\stable\package\60b04a80ac4dd5c23f1acffacc679e9902e636e5\share\cmake-3.15\Modules"
export CMAKE_MODULE_PATH
PATH="C:\Users\jbounthong\.conan\data\cmake_installer\3.15.4\conan\stable\package\60b04a80ac4dd5c23f1acffacc679e9902e636e5\bin"${PATH+:$PATH}
export PATH

$ source activate.sh

(conanenv) 
$ cmake --version
cmake version 3.15.4

(conanenv)
$ where cmake
C:\Users\jbounthong\.conan\data\cmake_installer\3.15.4\conan\stable\package\60b04a80ac4dd5c23f1acffacc679e9902e636e5\bin\cmake.exe
C:\Program Files\CMake\bin\cmake.exe
```

You may want to elaborate on you environment, it may be a slightly different issue, in which case would warrant filing it as a different bug.

**Comment by memsharded** [comment]
Merged to release/1.21.1 branch, will be released next week. Thanks for all the feedback!
